### DISCUSSION TITLE
如何在其他窗体关闭另一个窗体的dialog

### DISCUSSION THREAD

**Discussion by frexspar** [HandyOrg/HandyControl#1331] [General]
A窗体Show出来的Dialog如果通过别的窗体调用Close来关闭?

**Comment by boogaloo-jw** [comment]
我是这么做的，你可以参考下。
A窗体记录Dialog的全局变量，比如DialogA=Dialog.Show(); B窗体做完了某件事之后，使用Messenger类进行消息通知A，Handycontrol中封装了这个类（MVVMToolKit、Prism中都有类似功能），窗体A收到消息后关闭窗口DialogA.Close()；Messenger类就相当于中介者，用于协调ViewModel之间的沟通

**Comment by NaBian** [comment]
使用 token
